modules/telemetry/src/metric_record.cpp

```diff
@@ -0,0 +1,152 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include "hephaestus/telemetry/metric_record.h"
+
+#include <cerrno>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <absl/base/thread_annotations.h>
+#include <absl/log/log.h>
+
+#include "hephaestus/concurrency/message_queue_consumer.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
+namespace heph::telemetry {
+
+namespace internal {
+namespace {
+[[nodiscard]] auto stringToInt64(const std::string& str) -> std::optional<int64_t> {
```

Comment:
Nit: I personally like to start functions with a verb to indicate action, but more a manner of taste than everything else

Reply:
good point, however for conversion methods we either omit the source as we can get it from the input or it becomes very verbose, like `convertStringToInt64`, no?

Reply:
yeah I see your point. I personally would prefer consistency over verbosity, also since this is only used in a small scope... but any version is fine by me